\documentclass[oupdraft]{sysbio_sse}
%\usepackage[colorlinks=true, urlcolor=citecolor, linkcolor=citecolor, citecolor=citecolor]{hyperref}
\usepackage{url}
\usepackage{indentfirst}
\usepackage{color}
\usepackage{float}
%\usepackage{hyperref}
% Add history information for the article if required
%\history{Received Month X, 20XX;
%revised Month X, 20XX}
%\hypersetup{linkcolor=black, citecolor=black, colorlinks=true, hyperindex=true}
\begin{document}

% Title of paper
\title{OpenTree: A Python package for Accessing and Analyzing data from the Open Tree of Life}
% Each important word in the title should begin with a capital letter

% List of authors, with corresponding author marked by asterisk
\author{Emily Jane McTavish$^{1,\ast}$, Luna Luisa Sanchez Reyes$^{1}$, and
Mark T. Holder,$^{3}$\\[4pt]
% Author addresses
\textit{$^{1}$~Dept.~Life and Environmental Sciences, University of California, Merced, CA, USA 95343}
\\
\textit{$^{2}$~Dept.~Ecology and Evolutionary Biology, University of Kansas, Lawrence, KS USA 66045}\\
\textit{$^{3}$~Biodiversity Institute, University of Kansas, Lawrence, KS USA 66045}
\\[2pt]
% E-mail address for correspondence
\textit{*Corresponding author details here}}
% Identify the name, address, telephone/fax numbers, and e-mail address for the author who will receive proofs and be designated the "corresponding author" in text.

% Running headers of paper:
\markboth%
% First field is the short list of authors
{McTavish, Sanchez-Reyes, Holder}
% Second field is the short title of the paper
{opentree Python package}
% This should be shortened version of the title and no greater than 50 characters

\maketitle

\begin{abstract}
{The Open Tree of Life project constructs a comprehensive, dynamic and digitally-available tree of life by synthesizing published phylogenetic trees along with taxonomic data.
Open Tree of Life provides web-service application programming interfaces (APIs) to make the tree estimate, unified taxonomy, and input phylogenetic data available to anyone.
Here, we describe the Python package \texttt{opentree}, which provides a user friendly Python wrapper for these APIs and a set of scripts and tutorials for straightforward downstream data analyses.
We demonstrate the utility of these tools by generating an estimate of the phylogenetic relationships of all bird families, and by capturing a phylogenetic estimate for all taxa  observed at the University of California Merced Vernal Pools and Grassland Reserve.
}

{Python, phylogenetics, taxonomy, evolution, open science.}
\end{abstract}
\newline

Evolutionary history provides a framework to link all life on earth. However, it is not easy to access accurate, up-to-date phylogenetic relationships for arbitrary sets of taxa of interest, even if phylogenetic estimates for those taxa have been made and published \citep{drew_lost_2013, mctavish_how_2017}. Individual phylogenetic estimates are not comprehensive, and therefore seldom contain all taxa of interest. Taxonomic relationships, while comprehensive, provide a coarse, and often outdated, picture of shared ancestry.
The Open Tree of Life project (OpenTree) provides a reproducible framework for accessing up-to-date and expert-knowledge-based evolutionary relationships for arbitrary sets of taxa across the entire tree of life.
All data in OpenTree is freely available via application programming interfaces (API's).
The package \texttt{opentree} provides a user friendly Python interface to access these data. In addition \texttt{opentree} is packaged with a set of tutorials and scripts to make common downstream analyses straightforward.

\bigskip
% Each important word in the heading level 1 should begin with a capital letter; no heading for the introduction
% Please note that the level 1 headings given here, e.g. Description, are suggestions only
\section{Description}
\label{sec2}

\texttt{opentree} is a Python package for accessing and analyzing data from the Open Tree of Life project (OpenTree).
OpenTree stores a wealth of taxonomic and phylogenetic data gathered together in an open-access interoperable framework.
The current OpenTree synthetic tree \citep{opentreeoflife_open_2019} comprises 2.4 million tips.
Most of the tips of this tree represent species, but some are infraspecific taxa.
The framework of the synthetic tree is provided by a unified taxonomy \citep{opentreeoflife_open_2019-1, rees_automated_2017}.
This taxonomy links unique identifiers across many online taxonomic resources, including but not limited to NCBI \citep{federhen_ncbi_2012}, and GBIF \citep{gbif_secretariat_gbif_2019}, as well as user contributed taxonomic amendments contained in [https://github.com/OpenTreeOfLife/amendments-1].
These taxonomic relationships are currently refined by evolutionary estimates from 1,216 published phylogenetic trees (phylogenies) encompassing 87,000 tip taxa \citep{opentreeoflife_open_2019, redelings_supertree_2017}.
The Open Tree of Life database, `Phylesystem' \citep{mctavish_phylesystem_2015} contains 4,500 published studies, including those incorporated in the latest version of OpenTree's synthetic tree. Phylogenies in the Phylesystem database have mappings between their tip names and unique OpenTree taxonomic identifiers.

All of these data are freely accessible via API calls, documented at \url{https://github.com/OpenTreeOfLife/germinator/wiki/Open-Tree-of-Life-Web-APIs}.
\texttt{opentree} provides a user-friendly wrapper for calling these APIs from the command line or in Python.
In addition, it converts these data between commonly used file formats and data types.
This package allows users to generate data objects to use in DendroPy, a widely cited Python phylogenetic computing library \citep{sukumaran_dendropy_2010}.


\texttt{opentree} incorporates in Python the functionality available in rotl, an {R} package to interact with the Open Tree of Life data \citep{michonneau_rotl_2016}, as well as additional downstream analysis and interoperability tools.
\texttt{rotl} has been already been cited 132 times in the 4 years since its publication, demonstrating a demand for accessible user access to these data.
By providing a Python package to interact with these data, we make it straightforward for Python users to access and analyze these data.
A Python wrapper for Open Tree of Life also makes linking these data with the stable of other Python biodiversity informatics tools straightforward.

Finally, \texttt{opentree} expands the toolset available for working with OpenTree's unified taxonomy \citep{rees_automated_2017}.


\bigskip
\section{Services provided by opentree}
\label{sec3}


The OpenTree APIs are divided into three main categories: synthetic tree, taxonomy and taxonomic name resolution, and study search.
Many analyses integrate calls to each of these subcomponents.
The \texttt{opentree} package links across these services to make common API calls easier.
Some example calls are described here, but all methods and scripts are fully documented, including examples and return formats at \url{https://opentree.readthedocs.io}.

\subsubsection{Synthetic tree.---} The OpenTree synthetic tree contains all 2.4 million taxa in the OpenTree taxonomy, with relationships for 87,000 taxa informed by 1,216 studies.
Each branch in the tree is informed by published phylogenetic relationships, where they are present in the curated data store, or by taxonomic relationships where no phylogenetic data is available.
For each node in the synthetic tree, the API returns identifiers for
all the phylogenies in the synthesis pipeline which support or conflict with that node.
Each node is uniquely labeled.
If the descendants of a node align with named taxonomic group, the taxon identifier is applied to the node.
If the node does not correspond to a taxon named in the OpenTree taxonomy, the node is labeled using a phyloreference \citep{parr_evolutionary_2012}, a unique name identifying that node as the most recent common ancestor of two named taxa.
\texttt{opentree} users can easily access evolutionary estimates for arbitrary sets of taxa.
The web-service response also includes the published phylogenetic estimates which underlie those inferences.
The \texttt{opentree} wrapper captures and formats these citations to make providing appropriate credit for these synthetic induced subtree estimates straightforward.
Users can also access full synthetic subtrees subtending any individual node.


\subsubsection{Taxonomy and Taxonomic Name Resolution.---} The OpenTree taxonomy not only provides a scaffold for the synthetic tree, but is also a valuable resource in its own right.
Matching names is a key hurdle in bioinformatics.
Correct taxon names change through time, and spelling and typographical errors can propagate through bioinformatic resources.
Thus, demanding exact matching of names from different sources can be too stringent and fail to match the same taxon.
However, different names can be very close in spelling to one another.
So, tolerating misspellings makes it easy to accidentally match names that should refer to two distinct taxa.
The OpenTree taxonomy \citep{rees_automated_2017, opentreeoflife_open_2019-1} provides a link between the unique identifiers generated by several large scale online taxonomic resources [GBIF, NCBI, Silva, Worms], as well as all known name synonomies provided by those resources.
The OpenTree taxonomic name resolution service (TNRS) searches this full taxonomy and returns exact or fuzzy matches from input names string to taxa and their unique taxonomic identifiers.
This TNRS forms a link between human readable name strings, and rigorous unique identifiers.

\texttt{opentree} wraps the OpenTree taxonomy and TNRS APIs for ease of integrating taxonomy and TNRS queries with downstream analyses. In addition, \texttt{opentree} provides helpers for quickly searching the text download of the taxonomy, which can be more efficient for bulk queries.


\subsubsection{Study search.---} The OpenTree datastore contains 4,468 published phylogenetic studies, including 9,395 phylogenies (as of Dec 4, 2020).
These studies and phylogenies are indexed on a number of properties including author name, curator name, and publication information.
In addition, the tips of these trees are mapped to the unified OpenTree taxonomy making comparisons among estimates of relationships and searching for taxa of interest straightforward.
This allows searching of studies based on taxa contained in the study.
Importantly, this search does not rely on string-matching of what the taxonomic name was at the time of publication -- rather, it leverages the full suite of synonomies gathered across the input taxonomies to find equivalent taxa across studies, even if the canonical name has changed between publications.
The indexing of these phylogenies is taxonomically explicit.
So, for example, a search for `canidae' will find phylogenies with taxa contained in the taxonomic group Canidae, even if the term `canidae' does not itself appear in the phylogeny or tips.
Based on the results of these searches, studies can be viewed in a browser on the OpenTree curator site, or the phylogenies themselves can be downloaded for comparisons or other downstream use.

In addition, as the tips of each study are mapped by curators to taxonomic identifiers in the OpenTree taxonomy, comparing the relationships represented in input phylogenies to taxonomic relationships and to taxonomy is straightforward. The browser based tree viewer has a graphical visualization of this concordance and conflict. \texttt{opentree} provides a wrapper for this conflict functionality, which makes it straightforward to assess what taxon definitions and evolutionary relationships a phylogenetic estimate agrees with and conflicts with. This functionality can also be applied to local phylogenies for which the tips have been matched to taxonomy. This allows users to assess concordance and conflict with previous inferences in pre-publication phylogenies, even without sharing them to the publicly available OpenTree database \citep{reyes_physcraper_2020, mctavish_phylesystem_2015}.



\bigskip

\section{Biological Examples}
\label{sec4}

There are a plethora of downstream applications of this linked set of resources.
We highlight two examples based on user queries.


\subsection{A phylogeny of all bird families}
A full Jupyter notebook tutorial demonstrating how to access a synthetic tree of all bird families is packaged with the software at \url{https://github.com/OpenTreeOfLife/python-opentree/blob/main/docs/notebooks/TreeOfBirdGenera.ipynb}.
Capturing evolutionary information at large scales is often simplified by using arbitrary taxonomic cutoffs.
While the OpenTree taxonomy is not rank focused, it does track rank information from component taxonomies.
By searching the OpenTree taxonomy for families in birds, we find that there are 390 listed bird families, 196 of which are included in the synthetic tree.
Groups are excluded from the synthetic tree for a few potential reasons, the most common of which is that all members of the group are extinct, and we have no curated published studies providing information about the correct evolutionary relationships.
Placements of fossil taxa based only on taxonomy tend to be unstable, and the OpenTree synthesis procedure excludes taxa if the taxon is not present in at least one  phylogenetic input.
These families can be included in later synthetic trees if new studies or phylogenies are added to the Phylesystem database, providing information on their relationships.
Other taxa are excluded from synthesis if issues have been raised about their taxonomic validity, such as if the name corresponds to a family that is `barren', i.e. it contains no species in the OpenTree taxonomy, or because the name was judged to be invalid by the OpenTree taxonomy merging software \citep{rees_automated_2017}.

If we request an induced subtree from OpenTree's synthetic tree for these 164 taxa, we get back an output subtree that has 150 tips. The return value also includes a list of non-monophyletic taxa.
Some of the non-monophyletic taxa map to internal nodes on our output subtree. In those cases, input phylogenies are telling us that these `families' are paraphyletic with respect to other families. Which studies contest the monophyly of a taxonomic clade can be easily accessed through the browser (e.g. https://tree.opentreeoflife.org/opentree/argus/ottol@603925) or via the \texttt{opentree} wrapper using queries to \texttt{opentree.synth\_subtree}. Figure \ref{birdfams} shows the topology of 130 monophyletic bird families plus MRCA's of 20 additional non-monophyletic families as tips. The other 14 taxa are non-monophyletic families for which the MRCA is an internal node on the output subtree.


\begin{figure}[!h]
\centering\includegraphics[width=\textwidth]{bird_fam_fig}

\caption{Phylogenetic relationships of 150 bird families based on the latest OpenTree synthetic tree (v12.3), showing the number of published phylogenetic studies which support and conflict with each branch. For families which are not monophyletic according to published phylogenies, tips for those families are labelled with 'MRCA of taxa in X family name'. Heat maps show the number of tip taxa descendants from each tip, in log scale on left, and in exact numbers on left. Branch colors show the number of input studies which support (left, green) or conflict with (right, red) each inferred branch in the synthetic tree. Branch lengths are arbitrary. A total of 64 published phylogenies underlie the relationships in this subtree (citations in supplemental information). Figure created using the interactive Tree Of Life (iTOL) v4 \citep{letunic_interactive_2019}}
\label{birdfams}
\end{figure}

These families contain from 1 to 920 total descendant tip taxa (species and subspecies).
Across all 150 families, 10,357 descendent tip taxa are captured by the relationships shown in this subtree.
Figure \ref{birdfams} displays the number of descendant taxa in each family as a heat map, with log of the number of descendants displayed on the left, and the actual number of descendants on the right.
This display demonstrates that the use of `families' is not a very even way to break up biodiversity across birds.

Following OpenTree's phylogenetic synthesis algorithm \citep{redelings_supertree_2017}, each branch in the synthetic tree is supported
by either taxonomy alone (where there are no input phylogenetic studies that traverse that branch), or by one or more input phylogenetic studies.
The source of support for each node in the synthetic tree can be interrogated using a \texttt{synth\_node\_info} function call from the \texttt{OpenTree} class in the package here presented.
While each branch must be supported by taxonomy or at least one input study from Phylesystem, where multiple inputs traverse a branch, there can be conflict among studies.
OpenTree's synthesis algorithm is greedy, and the synthetic tree will display the branch supported by the highest ranked study included in synthesis.
The \texttt{synth\_node\_info} function will return not only which studies support any branch, but also which studies have relationships which conflict with that branch.
In figure \ref{birdfams}, support or conflict for each branch is displayed by the intensity of green and red coloration, respectively. Some branches in this subtree are supported by 20 studies, and a few show conflict with up to 17 other studies. Of the 443 branches in this subtree, 422 are supported by at least one input phylogenetic study, and the other 21 are based on taxonomic relationships.

It is important to note that OpenTree's synthetic tree shows only topology. When combining taxonomy, and phylogenetic branches from across studies with vastly different data types, merging branch lengths is not meaningful. However, estimates of node ages can be gathered using downstream tools such as DateLife \citep{sanchez-reyes_datelife_2019} which gathers and synthesizes node date information from studies in the Phylesystem database.

\subsection{Linking data from the Global Biodiversity Information Facility (GBIF) with phylogenetic information from Open Tree of Life}

\bigskip

The University of California (UC), Merced has a natural reserve directly adjacent to campus, which contains several vernal pools. These vernal pools create a unique habitat which allows native species to thrive, and the proximity to campus allows undergraduate classes to experience this ecosystem on field trips which can be accomplished during class time.
A species list for the reserve and adjacent campus areas is available through the Global Biodiversity Information Facility (GBIF) website \citep{gbif_secretariat_gbif_2019}. GBIF provides a repository for species occurrence data tracked in a variety of data stores, including bird observations from eBird \citep{sullivan_ebird_2009}, community science observations from iNaturalist (\url{www.inaturalist.org}), and several other resources. A full tutorial demonstrating how to access a subtree for a GBIF data download is included with this package at \url{https://github.com/OpenTreeOfLife/python-opentree/blob/main/docs/notebooks/gbif/GBIF_to_OpenTree.ipynb}.

We downloaded the full list of animal observations from the UC Merced Vernal pools reserve from GBIF \citep{gbif_secretariat_gbif_2019}. This data download comprised 6,709 records from 223 species. Using the GBIF unique taxon identifiers, 201 of these species could be directly matched to taxa in the OpenTree taxonomy using \texttt{opentree.taxon\_info(source\_id = {gbif unique identifier})}. This direct matching captures exact one to one relationships between these taxonomies, and avoids slow and potentially error prone string matching. Nineteen taxa had updated identifiers in GBIF since the most recent reconciliation between the GBIF taxonomy and the OpenTree taxonomy, and were assigned OpenTree taxon identifiers based on exact string matches. There were two taxa ``\textit{P. abortivum} St.'' and ``\textit{Ichneumon cupitus} Cresson 1877'', which were not found in the OpenTree taxonomy, and were dropped from the analysis.


Using this set of 223 OpenTree unique identifiers, an induced synthetic tree for these taxa can be downloaded (Figure \ref{vernalanimals}). This synthetic tree is supported by 160 individual published trees (citations in supplemental information).

\begin{figure}[!h]
\centering\includegraphics[width=\textwidth]{vernal_animals}
\caption{Evolutionary relationships between all animal taxon records in the UC Merced Vernal Pools and Grassland Reserve. Branch lengths are arbitrary. A total of 160 published phylogenies underlie the relationships in this tree (citations in supplemental information). Figure created using \citep{letunic_interactive_2019}}
\label{vernalanimals}
\end{figure}



For researchers, working in the vernal pools reserve, this subtree also provides the necessary information for community phylogenetic analyses. \citet{li_for_2019} demonstrated that synthetic phylogenies from the OpenTree project perform well in community phylogenetic studies. By providing ready access to these estimates, based on 160 previously published phylogenies, \texttt{opentree} makes basing ecological analyses in an accurate evolutionary framework straightforward.


The ability to build a phylogeny of local taxa is also a valuable pedagogical tool. One of us (EJM) used this phylogeny to discuss the diversity of life of animal life as part of a class exercise on vernal pools ecology and evolution, in an undergraduate evolution class.
Students visited the UC Merced Vernal Pools and Grassland Reserve, and then explored the evolutionary relationships of all the animal species recorded as observed in the reserve.
There are several threatened and endangered species on the vernal pools reserve, including two species of fairy shrimp, \textit{Branchinecta lynchii} (threatened) and \textit{Branchinecta mesovallensis} (endangered).
By working with a subtree of taxa found on and around campus, tree thinking examples in class can have a direct connection for students. For example, this subtree (Figure \ref{vernalanimals}) shows that the genus of dabbling ducks, \textit{Anas}, does not form a monophyletic group. Walking the tree of life has been demonstrated to be an effective way to get students to understand the connections among different lineages of life on earth \citep{ballen_walking_2017}. Walking through this subtree, and labelling major animal groups allows students to connect to the diversity of animal life based on the actual species surrounding them, rather than arbitrary textbook examples.

\bigskip
\section{Discussion}
\label{sec5}

\subsection{Comparison to other tools}

Tools that generate phylogenies from DNA sequence data mined from the GenBank genetic
database \citep{smith2009mega, sanderson2008phylota, antonelli2014supersmart}
have been widely used in the past decade. Still, the full phylogenetic history of the
tree of life has not been attained.
Generating good quality phylogenies \textit{de novo} using genetic data requires
significant expert knowledge about phylogenetic methods and of
the organism of study, as well as significant training time. Meanwhile, there are
already thousands of expert-based-knowledge peer-reviewed and published phylogenies
that, if made available and accessible, can be reused and reanalyzed for many purposes,
from scientific discovery in ecology and evolution to education, conservation and
outreach \citep{stoltzfus2013phylotastic, mctavish_phylesystem_2015, li2019common}.

TreeBASE \citep{piel2000treebase} is a public resource that stores published
phylogenies and underlying data with original taxon labels.
All data in TreeBASE is downloadable in computer readable format for reuse and reanalysis.
Phylogenies are searchable using study author names, study citations, accession numbers and taxon names.
TreeBASE does not have a taxonomic classification system, which limits
taxon name searches to exact matches of original study taxon labels \citep{anwar2009improved}.
This misses many or all trees containing ancestor or descendant taxa of the search taxonomic
name, especially when the taxon name used for search is a higher taxa \citep{chen2008phylofinder}.

% tools to integrate NCBI's taxonomy to improve taxonomic name searches on TreeBASE
% exist \citep{chen2008phylofinder, anwar2009improved, ranwez2009phyloexplorer}
% but were never integrated to TreeBASE and are not available as web services.
% OpenTree's taxonomy supports searches of phylogenies using taxon names
% that will correct misspellings and found synonomies, as well as ancestors or
% descendants of the search taxon.

TreeBASE does not have the possibility of extracting phylogenies with a
subset of taxa, or combining phylogenies with some or no overlapping taxa.
requiring further processing of phylogenies by the users if their study requires a phylogeny
with complete taxon sampling, a phylogeny containing a specific subset of taxa,
 or combining phylogenies (to match tips in a trait dataset for a comparative study, for example).

Dryad is also a repository of phylogenetic information, but it suffers from
the same shortcomings as TreeBASE, as it lacks a taxonomic classification.
Furthermore, trees are not always fully available as described because data is not curated.

STbase \citep{mcmahon2015stbase} is a project that constructed 1 million trees of
all eukaryotes using sequence data for use on comparative analyses.
The platform was meant to allow searching the
trees in a taxonomically explicit way, as well as obtaining subtrees of custom sets of taxa.
Unfortunately the website and the trees are not available at the moment of writing of
this manuscript, which could explain the small amount of citations it has (8 as of April 2021).

When studies require a particular set of taxa not represented in the same phylogeny,
or require complete taxon sampling, trees generated from taxonomic data have been
used for analyses \citep{tree2014tree}.
 However this is not ideal, especially when at least some phylogenetic information
 can be used to resolve taxonomy.
 % I need to look for more info on this, I think aybe Phylomatic allows this, I think.

Resources that combine taxonomic and phylogenetic knowledge:
The Angiosperm Phylogeny website \citep{stevens2001angiosperm} is a very valuable resource because it combines
available taxonomic and published phylogenetic information with the main goal to
portray a phylogenetic context for all flowering and seed plants. Information is
combined based on decisions and arguments from the author, that, while fully documented
on the website, is not an informatically reproducible process. The phylogenies are
also only available as image files, leaving users to the task of obtaining the phylogenies
in a computer readable format, which, in many cases, are only available from the
original authors upon request. The probability of success for that is only 30% (citation)

Resources that allow subsetting large phylogenies:

Resources such as fish tree of life \citep{chang2019r} and vertlife (https://vertlife.org/)
allow subsetting phylogenies of major taxonomic groups for fishes, mammals, amphibians, squamates and birds.
These phylogenies have been constructed based on genetic data and with branches
from missing taxa placed upon the phylogenetic backbone based on taxonomy. Fish
tree of life only allows obtaining trees corresponding to families, while vertlife
allows subsetting any combination of species, but not tips at higher taxonomic levels.
The subtrees that can be obtained with these resources are limited in taxonomic scope.

None of these resources combine the information that they have into a single tree.
In this sense, OpenTree is filling an empty niche in phylogenetics:
It combines phylogenetic information from across all taxonomic groups of life,
filling gaps in phylogenetic knowledge with expert taxonomic knowledge. The whole
tree is downloadable in computer readable format, which means it is not only available
but also accessible for reuse.

Having all information integrated into a single evolutionary context, OpenTree
is the only tool that can generate expert knowledge based trees of any combination
of taxa, at any taxonomic level.


\subsection{Application of OpenTree services}

Comparative biology necessitates a phylogenetic context to make sense of biological
observations and better understand biological patterns and processes
\citep{felsenstein1985phylogenies, soltis2003role}.
Ideally, evolutionary and ecological analyses require a dated phylogeny, with branch lengths
proportional to time, inferred using properly curated fossil calibrations applied
to a phylogeny reconstructed from carefully chosen genetic markers \citep{webb2008phylocom, jantzen2019effects}.
Still, nowadays, obtaining such good quality phylogenies requires
considerable more effort, resources, and expertise than reusing an expert-knowledge-based
published phylogenies \citep{li2019common}.

Phylogenies with and without branch lengths generated from
existing phylogenetic knowledge have been widely and successfully used in the field
for various studies.

OpenTree's synthetic tree in particular has been used to test the effect of
different types of branch lengths in phylogenies used to perform ecological community
analyses and to estimate phylogenetically based spatial biodiversity metrics
\citep{allen2019spatial, li2019common}.
\citep{allen2019spatial} tested different synthetic trees without branch lengths,
including OpenTree's synthetic tree, and concluded that they can be used successfully
to calculate phylogenetic distance and relative phylogenetic diversity to understand
ecological and evolutionary processes that can guide conservation efforts.
\citep{li2019common} dated OpenTree's synthetic tree using averaged age data from
the TimeTree database \citep{kumar2017timetree} as fixed node ages to evenly distribute ages of remaining
nodes using the BLADJ algorithm \citep{webb2008phylocom}. They concluded that phylogenetic diversity
and phylogenetic signal estimates from OpenTree's synthetic tree are as good
as those calculated from fossil-calibrated phylogenies based on DNA sequence data,
and suggest using the former for community ecology phylogenetic analyses when the
latter are unavailable \citep{li2019common}.

Comparative biology studies rely on trait datasets to contextualize them phylogenetically
and properly assess traits' evolutionary patterns and processes.
Unless an exceptional research planning phase happens early at the onset of a project,
most trait datasets are gathered without taking into account available phylogenetic information.
Taxa in biological trait datasets are usually non-overlapping with existing phylogenies,
and sparsely distributed across taxonomic groups that are rarely represented within
a single phylogeny.

OpenTree's subtree services allow getting a phylogeny for a targeted set of taxa
that can be distributed across distantly related biological groups.
OpenTree's taxon name matching services applied to taxon names in trait datasets
allow direct connection of trait data with phylogenetic trees in OpenTree's databases
and with the synthetic tree itself.

These OpenTree services have been used to address interesting biological
questions with comparative analyses, encompassing distantly related biological groups across the Tree of Life
\citep{tarka2018sex, healy2019animal, herrera2015predicting, capdevila2020longevity, fisher2017evolution, boeckmann2015quest, uyeda2017evolution}.

%% I think the following studies can be summarized into a single paragraph emphasizing
%% that they were applied to distantly related groups rarely represented within the same phylogeny
\citep{tarka2018sex} studied the evolution of sex differences in life history, behavior
and physiological traits using an OpenTree's synthetic subtree of 70 metazoan species (vertebrates and invertebrates),
encompassing marine polychaete worms, crayfish, spiders, amphibians, birds and mammals.
% with branch lengths estimated using Grafen 1989 method assuming node heights were raised to the power of 0.5.

\citep{healy2019animal} studied the phylogenetic distribution of life history strategies across metazoans
using an OpenTree's synthetic subtree of 121 species including corals, clams,
fish, crocodiles, birds and mammals.
% they grafted dated phylogenies of fish, reptiles birds and mammals and calculated the remaining branch lengths with Grafen's method.

\citep{capdevila2020longevity} examined differences in life history strategies between
terrestrial and aquatic species using an OpenTree's synthetic subtree of 807 species,
encompassing land plants, algae and animals.
% branch lengths estimated with ape::compute.brlen R function using Grafen's method.

\citep{herrera2015predicting} used an OpenTree's synthetic subtree of 434 species
including fungi, metazoans and plants to predict RAD-seq marker cleavage site numbers
across the eukaryotic tree of life to guide the design of genome-wide genotypic and
sequencing projects.

\citep{boeckmann2015quest} used an OpenTree's synthetic tree of 147 species encompassing
prokaryotes and eukaryotes as a species tree hypothesis to understand the evolution
of gene families and taxa to improve orthology prediction for phylogenetic applications.
% they used topology only.


\citep{fisher2017evolution} used an OpenTree subtree of 89 species including insects, plants,
fungi, molluscs, arachnids and worms, to study the evolution of host-symbiont
dependence across the tree of life.
% cited 117 times; used Bayesian phylogenetic mixed model to account for phylogenetic effects,
% seems like tree has no branch lengths, but data available upon request :/

\citep{kunc2019effects} got OpenTree synthetic subtrees for arthropods, molluscs,
amphibians, mammals, fish and birds to test the effect of anthropogenic noise on
animal behavior. They found all groups display alterations to their behavior, meaning
that anthropogenic noise is not affecting any taxonomic groups selectively.
% cited 21 times; obtained branch lengths using Grafen's method, too



--------------------------------
Community ecology and comparative studies often require phylogenetic context for thousands of
species belonging to distantly related taxonomic groups across the tree of life \citep{}.
Sometimes, constructing a tree from available genetic data from GenBank for a group
is unfeasible due to the large amount of samples required.
Other times, available genetic data on GenBank is not enough to address the goals
of a study \citep{owen2015synthetic, brown2017development, ewers2019towards}.
For example, broad taxon sampling coverage is key to correctly estimate phylogenetic
diversity \citep{jantzen2019effects, park2018taxon}, and genetic data is usually
not available for all taxa in a community or a biological group of study.
% in \citep{bitomsky2020clade}


Researchers have gathered published phylogenies reconstructed with genetic data
and used OpenTree's services to standardize taxon names to OpenTree's reference taxonomy,
synthesize the phylogenetic data with taxonomic data using OpenTree's synthesis algorithm,
and generate a fully sampled synthetic tree for their taxa of interest
\citep{owen2015synthetic, brown2017development, ewers2019towards}.

\citep{owen2015synthetic} generated a 387 tip phylogeny of freshwater crayfish using DNA sequences
data mined from GenBank with PHLAWD \citep{smith2009mega}. Then, they used OpenTree's services
 to standardize that and 19 previously published phylogenies
reconstructed from genetic data, with diverse taxon sampling within the freshwater crayfish
 to Opentree's taxonomy, to generate an OpenTree synthetic tree for all 590 species
of freshwater crayfish and 129 additional subspecies.
% It would be superfun to know how many of these branches are supported by phylogenetic
% data and how many by taxonomy, so maybe we can say that sometimes mining sequence data
% from GenBank is not enough to get the most phylogenetic data of a group, when genetic
% data is not released on GenBank, for example.
% owen et al. 2015 has been cited 32 times
With the synthetic subtree they were able to fully map IUCN endangered species
status in crayfish, test for associations with clades and taxonomy, and improve conservation
strategies in the group. They wouldn't have been able to do this with the DNA based phylogeny
because it has incomplete sampling.
They also used the conflict application in the curator app to detect conflict between
taxonomy and their phylogenetic estimates based on genetic data, allowing them to
discuss and identify regions that require more sequencing effort.

\citep{brown2017development} gathered 290 published bird phylogenies based on genetic
data, some of them with broad but incomplete taxonomic sampling (Jetz et al.????),
to use as input to generate an OpenTree synthetic subtree of birds.
They then used OpenTree's conflict application to evaluate node concordance and
conflict across the input phylogenies and the synthetic tree. They found that early diverging nodes
 seem to have the more conflict, informing future research sequencing efforts to
 focus on getting markers to resolve those areas of the bird phylogeny.

\citep{ewers2019towards} added 27 phylogenies of barnacles based on genetic data
(the best sampled one encompassing 124 species) to OpenTree's database and generated
a 2,070 synthetic tree of barnacle species and subspecies, containing phylogenetic
data for 239 species, as opposed to the 124 species in the best sampled available
phylogeny based on genetic data alone. They mapped geographic occurrence data unto
the phylogeny and detected that phylogenetic information is especially lacking for
deep sea and Arctic taxa.

Moreover, having a larger phylogenetic framework allows tackling taxonomic questions,
guiding future phylogenetic efforts and including large numbers of species into
comparative studies \citep{ewers2019towards}.

Researchers have used subtrees derived from OpenTree's synthetic tree encompassing species present
in specific ecological communities or encompassing distantly related groups, to
use as backbone to estimate branch lengths
based on genetic data \citep{allen2019spatial, smith2018constructing}, or using
age data from dated phylogenies in the OpenTree database \citep{uyeda2017evolution, geffroy2020evolutionary}.

\citep{uyeda2017evolution} used an OpenTree subtree of mammals and dated it using OpenTree
chronograms. They then grafted it to published chronograms from birds, amphibians
and reptiles into a single dated synthetic tree of 837 vertebrate species to study
the evolution of metabolic energetic scaling in order to improve macroecological predictions.
% branch lengths are proportional to time; they tested a model of trait evolution.

\citep{geffroy2020evolutionary} dated an OpenTree synthetic subtree gathering age data
from dated phylogenies stored in OpenTree's Phylesystem database. They found that
increased interactions with humans (indiced by captivity, urbanization, or domestication)
reduce the overall mean of antipredator traits over time in a sample of 102 species
of mammals, birds, reptiles, fish, amphibians, arthropods, and mollusks.
% has been cited 6 times; used OpenTree synthetic tree and dated it using chronograms in Phylesystem.



I also wanna say that Phylomatic \citep{webb2005phylomatic} is a tool that came before
OpenTree was available, and it has been used for a wealth of plant community ecology
studies (has been cited more than 1000 times).
Phylomatic is still being used (cited more than 100 times since 2020) for mammals
and plants because it is the only taxa available in there now.
Studies have used Phylomatic trees as backbone and then dated them. One study used
age data from chronograms in OpenTree to date a Phylomatic tree \citep{}.
OpenTree ha sthe potential to be used for bacterial and fungal community ecology studies.

In the absence of branch length data, phylogenies with arbitrary branch lengths
are a useful phylogenetic background to take into account phylogenetic structure
\citep{garland1992procedures}.
% I probably need a more recent citation besides garland 1992.
% all comparative studies that used opentree's tree with branch lengths obtained
% using Grafen's method can go in here!

\citep{jantzen2019effects} suggests that using OpenTree's synthetic subtree with
branch lengths inferred from additional genetic data or time calibrating them directly
will facilitate many more ecological community phylogenetic analyses, as it co-opts existing
phylogenetic knowledge and can represent phylogenetic relationships of distantly related organisms.



\section{Availability}
\label{sec6}

\texttt{opentree} is fully open source with a CC0 license. It is available on GitHub \url{ https://github.com/OpenTreeOfLife/python-opentree}. It can be installed from PyPi using \texttt{pip install opentree}. The code is packaged with an automated test suite which is maintained to cover at minimum 75\% of the code. Testing reports are generated on \url{travis.io} and posted to {codecov.io} and reflected on the GitHub readme with each commit. Documentation and tutorials are available with the code, and are posted to \url{https://opentree.readthedocs.io}.


\section{Conflict of Interest}
None

\section{Funding}
Funding for this project was provided by NSF ABI 1759846 and NSF ABI 1759838.

%If you have any acknowledgements, please include them here.
\section{Acknowledgements}
This package relies on the OpenTree API's and the team that continues to develop and improve them, currently Jim Allman, Karen Cranston, Ben Redelings, and the authors of this package. We acknowledge logistical support provided by the UC Merced Vernal Pools and Grassland Reserve.
Thank you to Emily Sessa and the organizers of the Society of Systematic Biologists SSB 2020 meeting for supporting the workshop where we tested and demonstrated this package.

%If your paper has accompanying supplementary data, please include the below statement in your PDF.
%\section{Supplementary Material}
%Data available from the Dryad Digital Repository:
% \href{http://dx.doi.org/10.5061/dryad.[NNNN]}%
% {http://dx.doi.org/10.5061/dryad.[NNNN]}.
%\url{http://dx.doi.org/10.5061/dryad.[NNNN]}.

\bigskip\bigskip

%%%%%%%%%%%%%%%%%%%% REFERENCES %%%%%%%%%%%%%%%%%%
\bibliographystyle{plainnat}
\bibliography{paper}
% 1. All authors should be listed i.e. no use of et al.
% 2. Dashes should not be used to replace author names in repeat entries
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

% Please include any figure captions on a separate page after the references. Figures themselves should be embedded in the text.

%\begin{figure}[!p]
%\centering\includegraphics{fig1}
%\caption{Figure caption}
%\label{Fig1}
%\end{figure}


%\begin{table}[!p]
% 1. Table titles should be in caps and lowercase
% 2. Footnotes can be used in Tables (a,b,c)}
%\tblcaption{Table title
%\label{Table1}}
%{\tabcolsep=4.25pt
%\begin{tabular}{@{}cccccccccc@{}}
%\tblhead{Heading & Heading & Heading & Heading & Heading}
%Value & Value & Value & Value & Value
%\lastline
%\end{tabular}}
%\end{table}

%If you have any print appendices, please include them at the end of the document.

\end{document}
