\documentclass[oupdraft]{sysbio_sse}
%\usepackage[colorlinks=true, urlcolor=citecolor, linkcolor=citecolor, citecolor=citecolor]{hyperref}
\usepackage{url}
\usepackage{indentfirst}
\usepackage{color}
\usepackage{float}
%\usepackage{hyperref}
% Add history information for the article if required
%\history{Received Month X, 20XX;
%revised Month X, 20XX}
%\hypersetup{linkcolor=black, citecolor=black, colorlinks=true, hyperindex=true}
\begin{document}

% Title of paper
\title{OpenTree: A Python package for Accessing and Analyzing data from the Open Tree of Life}
% Each important word in the title should begin with a capital letter

% List of authors, with corresponding author marked by asterisk
\author{Emily Jane McTavish$^{1,\ast}$, Luna Luisa Sanchez Reyes$^{1}$, and
Mark T. Holder,$^{3}$\\[4pt]
% Author addresses
\textit{$^{1}$~Dept.~Life and Environmental Sciences, University of California, Merced, CA, USA 95343}
\\
\textit{$^{2}$~Dept.~Ecology and Evolutionary Biology, University of Kansas, Lawrence, KS USA 66045}\\
\textit{$^{3}$~Biodiversity Institute, University of Kansas, Lawrence, KS USA 66045}
\\[2pt]
% E-mail address for correspondence
\textit{*Correspondence to be sent to: Science and Engineering Building 1, University of California, Merced, 5200 N.
Lake Rd, Merced CA 95343 USA\\
E-mail: ejmctavish@ucmerced.edu
}}
% Identify the name, address, telephone/fax numbers, and e-mail address for the author who will receive proofs and be designated the "corresponding author" in text.

% Running headers of paper:
\markboth%
% First field is the short list of authors
{McTavish, Sanchez-Reyes, Holder}
% Second field is the short title of the paper
{opentree Python package}
% This should be shortened version of the title and no greater than 50 characters

\maketitle

\begin{abstract}
{The Open Tree of Life project constructs a comprehensive, dynamic and digitally-available tree of life by synthesizing published phylogenetic trees along with taxonomic data.
Open Tree of Life provides web-service application programming interfaces (APIs) to make the tree estimate, unified taxonomy, and input phylogenetic data available to anyone.
Here, we describe the Python package \texttt{opentree}, which provides a user friendly Python wrapper for these APIs and a set of scripts and tutorials for straightforward downstream data analyses.
We demonstrate the utility of these tools by generating an estimate of the phylogenetic relationships of all bird families, and by capturing a phylogenetic estimate for all taxa  observed at the University of California Merced Vernal Pools and Grassland Reserve.
}

{Python, phylogenetics, taxonomy, evolution, open science.}
\end{abstract}
\newline

Evolutionary history provides a framework to link all life on earth. However, it is not easy to access accurate, up-to-date phylogenetic relationships for arbitrary sets of taxa of interest, even if phylogenetic estimates for those taxa have been made and published \citep{drew_lost_2013, mctavish_how_2017}. Individual phylogenetic estimates are not comprehensive, and therefore seldom contain all taxa of interest. Taxonomic relationships, while comprehensive, provide a coarse, and often outdated, picture of shared ancestry.
The Open Tree of Life project (OpenTree) provides a reproducible framework for accessing up-to-date and expert-knowledge-based evolutionary relationships for arbitrary sets of taxa across the entire tree of life.
All data in OpenTree is freely available via application programming interfaces (API's).
The package \texttt{opentree} provides a user friendly Python interface to access these data. In addition \texttt{opentree} is packaged with a set of tutorials and scripts to make common downstream analyses straightforward.

\bigskip
% Each important word in the heading level 1 should begin with a capital letter; no heading for the introduction
% Please note that the level 1 headings given here, e.g. Description, are suggestions only
\section{Description}
\label{sec2}

\texttt{opentree} is a Python package for accessing and analyzing data from the OpenTree of Life project.
Open Tree of Life stores a wealth of taxonomic and phylogenetic data gathered together in an open-access interoperable framework.
The current synthetic tree \citep{opentreeoflife_open_2019} comprises 2.4 million tips.
Most of the tips of the tree represent species, but some are infraspecific taxa.
The framework of this tree is provided by a unified taxonomy \citep{opentreeoflife_open_2019-1, rees_automated_2017}.
This taxonomy links unique identifiers across many online taxonomic resources, including NCBI \citep{federhen_ncbi_2012}, GBIF \citep{gbif_secretariat_gbif_2019}, as well as user contributed taxonomic amendments contained in [https://github.com/OpenTreeOfLife/amendments-1].
These taxonomic relationships are refined by evolutionary estimates from 1,216 published papers including 87,000 tip taxa \citep{opentreeoflife_open_2019, redelings_supertree_2017}.
The Open Tree data store, `Phylesystem' \citep{mctavish_phylesystem_2015} contains 4,500 published studies, including those incorporated in the latest version of OpenTree's synthetic tree.
Phylogenies in the Phylesystem database have mappings between their tip names and unique OpenTree taxonomic identifiers.
There are several reasons trees are included in the `Phylesystem' data store but not the synthetic tree.
The complete curation of trees requires human intervention and confirmation at several steps, including vetting taxonomic name resolution service matches and correctly rooting trees, as many output files are shared as unrooted trees.
In addition, some of the trees in Phylesystem describe gene tree level or within species relationships, which are not appropriate for the species (and named subspecies) level relationships captured in the synthetic tree.
There is also lag between study upload, which makes studies immediately available via the API, and the time to re-build the entire synthetic tree, which is released in versioned increments (current version, 12.3)
The number of input studies increases in each successive synthesis tree release.

All of these data and tools are freely accessible via API calls, documented at \url{https://github.com/OpenTreeOfLife/germinator/wiki/Open-Tree-of-Life-Web-APIs}.
\texttt{opentree} provides a user-friendly wrapper for calling these APIs from the command line or from a Python interactive session.
In addition, it converts OpenTree data between commonly used file formats and data types.
This package allows users to generate data objects for use in DendroPy, a widely cited Python phylogenetic computing library \citep{sukumaran_dendropy_2010}.


\texttt{opentree} incorporates in Python the functionality available in rotl, an {R} package to interact with the Open Tree of Life data \citep{michonneau_rotl_2016}.
Additionally, \texttt{opentree} expands the toolset available for working with OpenTree's unified taxonomy \citep{rees_automated_2017}, as well as additional downstream analysis and interoperability tools.

\texttt{rotl} has already been cited 132 times in the 4 years since its publication, demonstrating a demand for accessible user access to OpenTree data and tools.
By providing a Python package to interact with OpenTree's data and services, we make it straightforward for Python users to access and analyze these data,
generally improving the accessibility of OpenTree.

The \texttt{opentree} Python package for OpenTree facilitates linking all these data with the stable of other Python biodiversity informatics tools.



\bigskip
\section{Services provided by opentree}
\label{sec3}


OpenTree APIs are divided into three main categories: synthetic tree, taxonomy and taxonomic name resolution, and study search.
Many analyses integrate calls from each of these categories.
The \texttt{opentree} package links across these services to facilitate calls to commonly used OpenTree APIs.
Some example calls are described here, but all methods and scripts are fully documented, including examples and return formats, at \url{https://opentree.readthedocs.io}.

\subsubsection{Synthetic tree.---} The OpenTree synthetic tree contains all 2.4 million taxa in the OpenTree taxonomy, with relationships for 87,000 taxa informed by 1,216 studies.
Each branch in the tree is informed by published phylogenetic relationships, where they are present in the curated data store, or by taxonomic relationships where no phylogenetic data is available.
For each node in the synthetic tree, the API returns identifiers for
all the phylogenies in the synthesis pipeline which support or conflict with that node.
Each node is uniquely labeled.
If the descendants of a node align with named taxonomic group, the taxon identifier is applied to the node.
If the node does not correspond to a taxon named in the OpenTree taxonomy, the node is labeled using a phyloreference \citep{parr_evolutionary_2012}, a unique name identifying that node as the most recent common ancestor of two named taxa.
\texttt{opentree} users can easily access evolutionary estimates for arbitrary sets of taxa.
The web-service response also includes the published phylogenetic estimates which underlie those inferences.
The \texttt{opentree} wrapper captures and formats these citations to make providing appropriate credit for these synthetic induced subtree estimates straightforward.
Users can also access full synthetic subtrees subtending any individual node.


\subsubsection{Taxonomy and Taxonomic Name Resolution.---} The OpenTree taxonomy not only provides a scaffold for the synthetic tree, but is also a valuable resource in its own right.
Matching names is a key hurdle in bioinformatics.
Correct taxon names change through time, and spelling and typographical errors can propagate through bioinformatic resources.
Thus, demanding exact matching of names from different sources can be too stringent and fail to match the same taxon.
However, different names can be very close in spelling to one another.
So, tolerating misspellings makes it easy to accidentally match names that should refer to two distinct taxa.
The OpenTree taxonomy \citep{rees_automated_2017, opentreeoflife_open_2019-1} provides a link between the unique identifiers generated by several large scale online taxonomic resources [GBIF, NCBI, Silva, Worms], as well as all known name synonomies provided by those resources.
The OpenTree taxonomic name resolution service (TNRS) searches this full taxonomy and returns exact or fuzzy matches from input names string to taxa and their unique taxonomic identifiers.
This TNRS forms a link between human readable name strings, and rigorous unique identifiers.

\texttt{opentree} wraps the OpenTree taxonomy and TNRS APIs for ease of integrating taxonomy and TNRS queries with downstream analyses. In addition, \texttt{opentree} provides helpers for quickly searching the text download of the taxonomy, which can be more efficient for bulk queries.


\subsubsection{Study search.---} The OpenTree datastore contains 4,468 published phylogenetic studies, including 9,395 phylogenies (as of Dec 4, 2020).
These studies and phylogenies are indexed on a number of properties including author name, curator name, and publication information.
In addition, the tips of these trees are mapped to the unified OpenTree taxonomy making comparisons among estimates of relationships and searching for taxa of interest straightforward.
This allows searching of studies based on taxa contained in the study.
Importantly, this search does not rely on string-matching of what the taxonomic name was at the time of publication -- rather, it leverages the full suite of synonomies gathered across the input taxonomies to find equivalent taxa across studies, even if the canonical name has changed between publications.
The indexing of these phylogenies is taxonomically explicit.
So, for example, a search for `canidae' will find phylogenies with taxa contained in the taxonomic group Canidae, even if the term `canidae' does not itself appear in the phylogeny or tips.
Based on the results of these searches, studies can be viewed in a browser on the OpenTree curator site, or the phylogenies themselves can be downloaded for comparisons or other downstream use.

In addition, as the tips of each study are mapped by curators to taxonomic identifiers in the OpenTree taxonomy, comparing the relationships represented in input phylogenies to taxonomic relationships and to taxonomy is straightforward. The browser based tree viewer has a graphical visualization of this concordance and conflict. \texttt{opentree} provides a wrapper for this conflict functionality, which makes it straightforward to assess what taxon definitions and evolutionary relationships a phylogenetic estimate agrees with and conflicts with. This functionality can also be applied to local phylogenies for which the tips have been matched to taxonomy. This allows users to assess concordance and conflict with previous inferences in pre-publication phylogenies, even without sharing them to the publicly available OpenTree database \citep{reyes_physcraper_2020, mctavish_phylesystem_2015}.



\bigskip

\section{Biological Examples}
\label{sec4}

There are a plethora of downstream applications of this linked set of resources.
We highlight two examples based on user queries.


\subsection{A phylogeny of all bird families}
A full Jupyter notebook tutorial demonstrating how to access a synthetic tree of all bird families is packaged with the software at \url{https://github.com/OpenTreeOfLife/python-opentree/blob/main/docs/notebooks/TreeOfBirdGenera.ipynb}.
Capturing evolutionary information at large scales is often simplified by using arbitrary taxonomic cutoffs.
While the OpenTree taxonomy is not rank focused, it does track rank information from component taxonomies.
By searching the OpenTree taxonomy for families in birds, we find that there are 390 listed bird families, 196 of which are included in the synthetic tree.
Groups are excluded from the synthetic tree for a few potential reasons, the most common of which is that all members of the group are extinct, and we have no curated published studies providing information about the correct evolutionary relationships.
Placements of fossil taxa based only on taxonomy tend to be unstable, and the OpenTree synthesis procedure excludes taxa if the taxon is not present in at least one  phylogenetic input.
These families can be included in later synthetic trees if new studies or phylogenies are added to the Phylesystem database, providing information on their relationships.
Other taxa are excluded from synthesis if issues have been raised about their taxonomic validity, such as if the name corresponds to a family that is `barren', i.e. it contains no species in the OpenTree taxonomy, or because the name was judged to be invalid by the OpenTree taxonomy merging software \citep{rees_automated_2017}.

If we request an induced subtree from OpenTree's synthetic tree for these 164 taxa, we get back an output subtree that has 150 tips. The return value also includes a list of non-monophyletic taxa.
Some of the non-monophyletic taxa map to internal nodes on our output subtree. In those cases, input phylogenies are telling us that these `families' are paraphyletic with respect to other families. Which studies contest the monophyly of a taxonomic clade can be easily accessed through the browser (e.g. https://tree.opentreeoflife.org/opentree/argus/ottol@603925) or via the \texttt{opentree} wrapper using queries to \texttt{opentree.synth\_subtree}. Figure \ref{birdfams} shows the topology of 130 monophyletic bird families plus most recent common ancestors (MRCA's) of 20 additional non-monophyletic families as tips. The other 14 taxa are non-monophyletic families for which the MRCA is an internal node on the output subtree.


\begin{figure}[!h]
\centering\includegraphics[width=\textwidth]{bird_fam_fig_rev}

\caption{Phylogenetic relationships of 150 bird families based on the latest OpenTree synthetic tree (v12.3). For families which are not monophyletic according to published phylogenies, tips for those families are labeled with `MRCA of taxa in X family name'. Heat maps show the number of tip taxa descendants in OpenTree within each tip. Branch colors show the number of input studies which support (left, green) or conflict with (right, red) each inferred branch in the synthetic tree. Branch lengths are arbitrary. A total of 64 published phylogenies underlie the relationships in this subtree (citations in supplemental information). Figure created using the interactive Tree Of Life (iTOL) v4 \citep{letunic_interactive_2019}}
\label{birdfams}
\end{figure}

These families contain from 1 to 920 total descendant tip taxa (species and subspecies).
Across all 150 families, 10,357 descendant tip taxa are captured by the relationships shown in this subtree.
Figure \ref{birdfams} displays the number of descendant taxa in each family as a heat map, with log of the number of descendants displayed on the left, and the actual number of descendants on the right.
This display demonstrates that the use of `families' is not a very even way to break up biodiversity across birds.

Based on OpenTree's phylogenetic synthesis algorithm \citep{redelings_supertree_2017}, each branch in the synthetic tree is supported
by either taxonomy alone (where there are no input phylogenetic studies that traverse that branch), or by one or more input phylogenetic studies.
The source of support for each node in the synthetic tree can be interrogated using a \texttt{synth\_node\_info} function call from the \texttt{OpenTree} class in the package here presented.
While each branch must be supported by taxonomy or at least one input study from Phylesystem, where multiple inputs traverse a branch, there can be conflict among studies.
OpenTree's synthesis algorithm is greedy, and the synthetic tree will display the branch supported by the highest ranked study included in synthesis.
The \texttt{synth\_node\_info} function will return not only which studies support a branch, but also which studies have relationships which conflict with that branch.
In figure \ref{birdfams} support or conflict for each branch is displayed by the intensity of green and red coloration, respectively. Some branches in this subtree are supported by 20 studies, and a few show conflict with up to 17 other studies. Of the 443 branches in this subtree, 422 are supported by at least one input phylogenetic study, and the other 21 are based on taxonomic relationships.

It is important to note that OpenTree's synthetic tree shows only topology. When combining taxonomy, and phylogenetic branches from across studies with vastly different data types, merging branch lengths is not meaningful. For downstream analyses requiring branch lengths, users have scaled topologies inferred from OpenTree using a variety of data types and rate-smoothing approaches e.g. \citep{smith2018constructing, smith2019pyphlawd, eastman2013congruification, allen2019spatial, li2019common, uyeda2017evolution, geffroy2020evolutionary, jantzen2019effects, sanchez-reyes_datelife_2019}.

\subsection{Linking data from the Global Biodiversity Information Facility (GBIF) with phylogenetic information from Open Tree of Life}

\bigskip

The University of California (UC), Merced has a natural reserve directly adjacent to campus, which contains several vernal pools. These vernal pools create a unique habitat which allows native species to thrive, and the proximity to campus allows undergraduate classes to experience this ecosystem on field trips which can be accomplished during class time.
A species list for the reserve and adjacent campus areas is available through the Global Biodiversity Information Facility (GBIF) website \citep{gbif_secretariat_gbif_2019}. GBIF provides a repository for species occurrence data tracked in a variety of data stores, including bird observations from eBird \citep{sullivan_ebird_2009}, community science observations from iNaturalist (\url{www.inaturalist.org}), and several other resources. A full tutorial demonstrating how to access a subtree for a GBIF data download is included with this package at \url{https://github.com/OpenTreeOfLife/python-opentree/blob/main/docs/notebooks/gbif/GBIF_to_OpenTree.ipynb}.

We downloaded the full list of animal observations from the UC Merced Vernal pools reserve from GBIF \citep{gbif_secretariat_gbif_2019}. This data download comprised 6,709 records from 223 species. Using the GBIF unique taxon identifiers, 201 of these species could be directly matched to taxa in the OpenTree taxonomy using \texttt{opentree.taxon\_info(source\_id = {gbif unique identifier})}. This direct matching captures exact one to one relationships between these taxonomies, and avoids slow and potentially error prone string matching. Nineteen taxa had updated identifiers in GBIF since the most recent reconciliation between the GBIF taxonomy and the OpenTree taxonomy, and were assigned OpenTree taxon identifiers based on exact string matches. There were two taxa ``\textit{P. abortivum} St.'' and ``\textit{Ichneumon cupitus} Cresson 1877'', which were not found in the OpenTree taxonomy, and were dropped from the analysis.


Using this set of 223 OpenTree unique identifiers, an induced synthetic tree for these taxa can be downloaded (Figure \ref{vernalanimals}). This synthetic tree is supported by 160 individual published trees (citations in supplemental information).

\begin{figure}[!h]
\centering\includegraphics[width=\textwidth]{vernal_animals}
\caption{Evolutionary relationships between all animal taxon records in the UC Merced Vernal Pools and Grassland Reserve. Branch lengths are arbitrary. A total of 160 published phylogenies underlie the relationships in this tree (citations in supplemental information). Figure created using using iTOL v4 \citep{letunic_interactive_2019}.}
\label{vernalanimals}
\end{figure}



For researchers, working in the vernal pools reserve, this subtree also provides the necessary information for community phylogenetic analyses. \citet{li_for_2019} demonstrated that synthetic phylogenies from the OpenTree project perform well in community phylogenetic studies. By providing ready access to these estimates, based on 160 previously published phylogenies, \texttt{opentree} makes basing ecological analyses in an accurate evolutionary framework straightforward.


The ability to build a phylogeny of local taxa is also a valuable pedagogical tool. One of us (EJM) used this phylogeny to discuss the diversity of life of animal life as part of a class exercise on vernal pools ecology and evolution, in an undergraduate evolution class.
Students visited the UC Merced Vernal Pools and Grassland Reserve, and then explored the evolutionary relationships of all the animal species recorded as observed in the reserve.
There are several threatened and endangered species on the vernal pools reserve, including two species of fairy shrimp, \textit{Branchinecta lynchii} (threatened) and \textit{Branchinecta mesovallensis} (endangered).
By working with a subtree of taxa found on and around campus, tree thinking examples in class can have a direct connection for students. For example, this subtree (Figure \ref{vernalanimals}) shows that the genus of dabbling ducks, \textit{Anas}, does not form a monophyletic group. Walking the tree of life has been demonstrated to be an effective way to get students to understand the connections among different lineages of life on earth \citep{ballen_walking_2017}. Walking through this subtree, and labeling major animal groups allows students to connect to the diversity of animal life based on the actual species surrounding them, rather than arbitrary textbook examples.

\bigskip
\section{Discussion}
\label{sec5}

The OpenTree project makes available a synthetic tree across 2.4 million taxa, as well as thousands of peer-reviewed and published phylogenies that together can be reused for applications
from scientific discovery to education, conservation and
outreach \citep{stoltzfus2013phylotastic, mctavish_phylesystem_2015, rosindell2012onezoom, wong2020dynamic}.
While several tools can generate phylogenies from DNA sequence data mined from the GenBank genetic database e.g. \citep{smith2009mega, sanderson2008phylota, antonelli2014supersmart, reyes_physcraper_2020}, for regions of the tree where phylogenetic estimates already exist, relying on published peer-reviewed inferences of relationships is more efficient and accurate than generating new estimates \citep{smith2018constructing, owen2015synthetic, brown2017development, ewers2019towards}.
Broad taxon sampling coverage is key to correctly estimate phylogenetic
diversity \citep{jantzen2019effects, park2018taxon}, and genetic data is usually not available for all taxa in a community or a biological group of study.

Many researchers have used the OpenTree's synthetic tree and its phylogenetic and taxonomic databases to answer their research questions across the breadth of both the tree of life, and of biological research domains.
In contrast to existing phylogenetic databases, \citep{piel2000treebase}, the links across studies provided by OpenTree and its synthesis algorithm make it possible to extract inferences for an arbitrary subset of taxa, and combine inferences from trees with few or no overlapping tips.
OpenTree services have been used to consolidate evolutionary estimates across taxa (freshwater crayfish \citep{owen2015synthetic}; birds \citep{brown2017development}; barnacles \citep{ewers2019towards, ewers2019testing}, mammals \citep{uyeda2017evolution}).
OpenTree's large scale synthetic subtrees have provide the opportunity to study
a variety of trait evolution topics such as the phylogenetic distribution of organism's life history traits at large scales \citep{tarka2018sex, healy2019animal, capdevila2020longevity}.
antipredator behavior associations with human interactions, \citep{geffroy2020evolutionary}, effects of anthropogenic noise on
animal behavior  \citep{kunc2019effects}, phylogenetic patterns of global wildlife trade,  \citep{fukushima2020global},  miniaturization in insects \citep{polilov2017scaling},
shoot flammability evolution \citep{cui2020shoot}, and host-symbiont dependence \citep{fisher2017evolution}.

In molecular biology, OpenTree's large scale synthetic subtrees
have been used to understand the evolution of gene families and taxa across prokaryotes and eukaryotes, to improve orthology prediction for phylogenetic applications  \citet{boeckmann2015quest}.
\citet{herrera2015predicting} used an OpenTree's synthetic subtree to predict RAD-seq marker cleavage site numbers
across the eukaryotic tree of life to guide the design of genome-wide genotypic and
sequencing projects. \citet{miller2020codonpairs} found that codon pairing biases are conserved across the diversity of the tree of life and can be used for phylogenomic analyses.

OpenTree's synthetic tree has been proved to provide a useful phylogenetic backbone for community phylogenetic analyses \citep{li2019common, jantzen2019effects}.
% who incorporated it to ecological Software?

% four outreach and education: onezoom and phylotastic

% other services for outreach and education: DateLife

% things that it can't do: topology search

This \texttt{opentree} Python toolkit expands user access to the OpenTree API's, and provides tutorials on how to easily access up-to-date evolutionary information across the entire tree of life, which will continue to facilitate biological research and outreach.


\section{Availability}
\label{sec6}

\texttt{opentree} is fully open source with a CC0 license. It is available on GitHub \url{ https://github.com/OpenTreeOfLife/python-opentree}. It can be installed from PyPi using \texttt{pip install opentree}. The code is packaged with an automated test suite which is maintained to cover at minimum 75\% of the code. Testing reports are generated on \url{travis.io} and posted to {codecov.io} and reflected on the GitHub readme with each commit. Documentation and tutorials are available with the code, and are posted to \url{https://opentree.readthedocs.io}.


\section{Conflict of Interest}
None

\section{Funding}
Funding for this project was provided by NSF ABI 1759846 and NSF ABI 1759838.

%If you have any acknowledgements, please include them here.
\section{Acknowledgements}
This package relies on the OpenTree API's and the team that continues to develop and improve them, currently Jim Allman, Karen Cranston, Ben Redelings, and the authors of this package. We acknowledge logistical support provided by the UC Merced Vernal Pools and Grassland Reserve.
Thank you to Emily Sessa and the organizers of the Society of Systematic Biologists SSB 2020 meeting for supporting the workshop where we tested and demonstrated this package.
We also appreciate the valuable comments of James Rosindell,
and an anonymous reviewer that greatly improved an earlier version of this manuscript.

%If your paper has accompanying supplementary data, please include the below statement in your PDF.
%\section{Supplementary Material}
%Data available from the Dryad Digital Repository:
% \href{http://dx.doi.org/10.5061/dryad.[NNNN]}%
% {http://dx.doi.org/10.5061/dryad.[NNNN]}.
%\url{http://dx.doi.org/10.5061/dryad.[NNNN]}.

\bigskip\bigskip

%%%%%%%%%%%%%%%%%%%% REFERENCES %%%%%%%%%%%%%%%%%%
\begin{thebibliography}{49}
\providecommand{\natexlab}[1]{#1}
\providecommand{\selectlanguage}[1]{\relax}
\providecommand{\bibAnnoteFile}[1]{%
  \IfFileExists{#1}{\begin{quotation}\noindent\textsc{Key:} #1\\
  \textsc{Annotation:}\ \input{#1}\end{quotation}}{}}
\providecommand{\bibAnnote}[2]{%
  \begin{quotation}\noindent\textsc{Key:} #1\\
  \textsc{Annotation:}\ #2\end{quotation}}

\bibitem[{Allen et~al.(2019)Allen, Germain-Aubrey, Barve, Neubig, Majure,
  Laffan, Mishler, Owens, Smith, Whitten et~al.}]{allen2019spatial}
Allen, J.~M., C.~C. Germain-Aubrey, N.~Barve, K.~M. Neubig, L.~C. Majure, S.~W.
  Laffan, B.~D. Mishler, H.~L. Owens, S.~A. Smith, W.~M. Whitten, et~al. 2019.
  Spatial phylogenetics of florida vascular plants: The effects of calibration
  and uncertainty on diversity estimates. IScience 11:57--70.
\bibAnnoteFile{allen2019spatial}

\bibitem[{Antonelli et~al.(2017)Antonelli, Hettling, Condamine, Vos, Nilsson,
  Sanderson, Sauquet, Scharn, Silvestro, Töpel, Bacon, Oxelman, and
  Vos}]{antonelli2014supersmart}
Antonelli, A., H.~Hettling, F.~L. Condamine, K.~Vos, R.~H. Nilsson, M.~J.
  Sanderson, H.~Sauquet, R.~Scharn, D.~Silvestro, M.~Töpel, C.~D. Bacon,
  B.~Oxelman, and R.~A. Vos. 2017. Toward a {Self}-{Updating} {Platform} for
  {Estimating} {Rates} of {Speciation} and {Migration}, {Ages}, and
  {Relationships} of {Taxa}. Systematic Biology 66:152--166.
\bibAnnoteFile{antonelli2014supersmart}

\bibitem[{Ballen and Greene(2017)}]{ballen_walking_2017}
Ballen, C.~J. and H.~W. Greene. 2017. Walking and talking the tree of life:
  {Why} and how to teach about biodiversity. PLOS Biology 15:e2001630.
\bibAnnoteFile{ballen_walking_2017}

\bibitem[{Boeckmann et~al.(2015)Boeckmann, Marcet-Houben, Rees, Forslund,
  Huerta-Cepas, Muffato, Yilmaz, Xenarios, Bork, Lewis
  et~al.}]{boeckmann2015quest}
Boeckmann, B., M.~Marcet-Houben, J.~A. Rees, K.~Forslund, J.~Huerta-Cepas,
  M.~Muffato, P.~Yilmaz, I.~Xenarios, P.~Bork, S.~E. Lewis, et~al. 2015. Quest
  for orthologs entails quest for tree of life: in search of the gene stream.
  Genome biology and evolution 7:1988--1999.
\bibAnnoteFile{boeckmann2015quest}

\bibitem[{Brown et~al.(2017)Brown, Wang, and Smith}]{brown2017development}
Brown, J.~W., N.~Wang, and S.~A. Smith. 2017. The development of scientific
  consensus: Analyzing conflict and concordance among avian phylogenies.
  Molecular phylogenetics and evolution 116:69--77.
\bibAnnoteFile{brown2017development}

\bibitem[{Capdevila et~al.(2020)Capdevila, Beger, Blomberg, Hereu, Linares, and
  Salguero-G{\'o}mez}]{capdevila2020longevity}
Capdevila, P., M.~Beger, S.~P. Blomberg, B.~Hereu, C.~Linares, and
  R.~Salguero-G{\'o}mez. 2020. Longevity, body dimension and reproductive mode
  drive differences in aquatic versus terrestrial life-history strategies.
  Functional Ecology 34:1613--1625.
\bibAnnoteFile{capdevila2020longevity}

\bibitem[{Cui et~al.(2020)Cui, Paterson, Wyse, Alam, Maurin, Pieper, Cubino,
  O’Connell, Donkers, Br{\'e}da et~al.}]{cui2020shoot}
Cui, X., A.~M. Paterson, S.~V. Wyse, M.~A. Alam, K.~J. Maurin, R.~Pieper, J.~P.
  Cubino, D.~M. O’Connell, D.~Donkers, J.~Br{\'e}da, et~al. 2020. Shoot
  flammability of vascular plants is phylogenetically conserved and related to
  habitat fire-proneness and growth form. Nature plants 6:355--359.
\bibAnnoteFile{cui2020shoot}

\bibitem[{Drew et~al.(2013)Drew, Gazis, Cabezas, Swithers, Deng, Rodriguez,
  Katz, Crandall, Hibbett, and Soltis}]{drew_lost_2013}
Drew, B.~T., R.~Gazis, P.~Cabezas, K.~S. Swithers, J.~Deng, R.~Rodriguez, L.~A.
  Katz, K.~A. Crandall, D.~S. Hibbett, and D.~E. Soltis. 2013. Lost branches on
  the tree of life. PLOS Biol 11:e1001636.
\bibAnnoteFile{drew_lost_2013}

\bibitem[{Eastman et~al.(2013)Eastman, Harmon, and
  Tank}]{eastman2013congruification}
Eastman, J.~M., L.~J. Harmon, and D.~C. Tank. 2013. {Congruification: Support
  for time scaling large phylogenetic trees}. Methods in Ecology and Evolution
  4:688--691.
\bibAnnoteFile{eastman2013congruification}

\bibitem[{Ewers-Saucedo et~al.(2019)Ewers-Saucedo, Owen, P{\'e}rez-Losada,
  H{\o}eg, Glenner, Chan, and Crandall}]{ewers2019towards}
Ewers-Saucedo, C., C.~L. Owen, M.~P{\'e}rez-Losada, J.~T. H{\o}eg, H.~Glenner,
  B.~K. Chan, and K.~A. Crandall. 2019. Towards a barnacle tree of life:
  integrating diverse phylogenetic efforts into a comprehensive hypothesis of
  thecostracan evolution. PeerJ 7:e7387.
\bibAnnoteFile{ewers2019towards}

\bibitem[{Ewers-Saucedo and Pappalardo(2019)}]{ewers2019testing}
Ewers-Saucedo, C. and P.~Pappalardo. 2019. Testing adaptive hypotheses on the
  evolution of larval life history in acorn and stalked barnacles. Ecology and
  evolution 9:11434--11447.
\bibAnnoteFile{ewers2019testing}

\bibitem[{Federhen(2012)}]{federhen_ncbi_2012}
Federhen, S. 2012. The {NCBI} {Taxonomy} database. Nucleic Acids Research
  40:D136--D143.
\bibAnnoteFile{federhen_ncbi_2012}

\bibitem[{Fisher et~al.(2017)Fisher, Henry, Cornwallis, Kiers, and
  West}]{fisher2017evolution}
Fisher, R.~M., L.~M. Henry, C.~K. Cornwallis, E.~T. Kiers, and S.~A. West.
  2017. The evolution of host-symbiont dependence. Nature Communications
  8:1--8.
\bibAnnoteFile{fisher2017evolution}

\bibitem[{Fukushima et~al.(2020)Fukushima, Mammola, and
  Cardoso}]{fukushima2020global}
Fukushima, C.~S., S.~Mammola, and P.~Cardoso. 2020. Global wildlife trade
  permeates the tree of life. Biological Conservation 247:108503.
\bibAnnoteFile{fukushima2020global}

\bibitem[{GBIF Secretariat(2019)}]{gbif_secretariat_gbif_2019}
GBIF Secretariat. 2019. {GBIF} occurence download. \url{https://doi.org/10.15468/dl.9bigak}
\bibAnnoteFile{gbif_secretariat_gbif_2019}

\bibitem[{Geffroy et~al.(2020)Geffroy, Sadoul, Putman, Berger-Tal, Garamszegi,
  M{\o}ller, and Blumstein}]{geffroy2020evolutionary}
Geffroy, B., B.~Sadoul, B.~J. Putman, O.~Berger-Tal, L.~Z. Garamszegi, A.~P.
  M{\o}ller, and D.~T. Blumstein. 2020. Evolutionary dynamics in the
  anthropocene: Life history and intensity of human contact shape antipredator
  responses. PLoS biology 18:e3000818.
\bibAnnoteFile{geffroy2020evolutionary}

\bibitem[{Healy et~al.(2019)Healy, Ezard, Jones, Salguero-G{\'o}mez, and
  Buckley}]{healy2019animal}
Healy, K., T.~H. Ezard, O.~R. Jones, R.~Salguero-G{\'o}mez, and Y.~M. Buckley.
  2019. Animal life history is shaped by the pace of life and the distribution
  of age-specific mortality and reproduction. Nature Ecology \& Evolution
  3:1217--1224.
\bibAnnoteFile{healy2019animal}

\bibitem[{Herrera et~al.(2015)Herrera, Reyes-Herrera, and
  Shank}]{herrera2015predicting}
Herrera, S., P.~H. Reyes-Herrera, and T.~M. Shank. 2015. {Predicting RAD-seq
  marker numbers across the eukaryotic tree of life}. Genome biology and
  evolution 7:3207--3225.
\bibAnnoteFile{herrera2015predicting}

\bibitem[{Jantzen et~al.(2019)Jantzen, Whitten, Neubig, Majure, Soltis, and
  Soltis}]{jantzen2019effects}
Jantzen, J.~R., W.~M. Whitten, K.~M. Neubig, L.~C. Majure, D.~E. Soltis, and
  P.~S. Soltis. 2019. Effects of taxon sampling and tree reconstruction methods
  on phylodiversity metrics. Ecology and evolution 9:9479--9499.
\bibAnnoteFile{jantzen2019effects}

\bibitem[{Kunc and Schmidt(2019)}]{kunc2019effects}
Kunc, H.~P. and R.~Schmidt. 2019. The effects of anthropogenic noise on
  animals: a meta-analysis. Biology letters 15:20190649.
\bibAnnoteFile{kunc2019effects}

\bibitem[{Letunic and Bork(2019)}]{letunic_interactive_2019}
Letunic, I. and P.~Bork. 2019. Interactive {Tree} {Of} {Life} ({iTOL}) v4:
  recent updates and new developments. Nucleic Acids Research 47:W256--W259.
\bibAnnoteFile{letunic_interactive_2019}

\bibitem[{Li et~al.(2019{\natexlab{a}})Li, Trotta, Marx, Allen, Sun, Soltis,
  Soltis, Guralnick, and Baiser}]{li2019common}
Li, D., L.~Trotta, H.~E. Marx, J.~M. Allen, M.~Sun, D.~E. Soltis, P.~S. Soltis,
  R.~P. Guralnick, and B.~Baiser. 2019{\natexlab{a}}. For common community
  phylogenetic analyses, go ahead and use synthesis phylogenies. Ecology
  100:e02788.
\bibAnnoteFile{li2019common}

\bibitem[{Li et~al.(2019{\natexlab{b}})Li, Trotta, Marx, Allen, Sun, Soltis,
  Soltis, Guralnick, and Baiser}]{li_for_2019}
Li, D., L.~Trotta, H.~E. Marx, J.~M. Allen, M.~Sun, D.~E. Soltis, P.~S. Soltis,
  R.~P. Guralnick, and B.~Baiser. 2019{\natexlab{b}}. For common community
  phylogenetic analyses, go ahead and use synthesis phylogenies. Ecology
  100:e02788.
\bibAnnoteFile{li_for_2019}

\bibitem[{McTavish et~al.(2017)McTavish, Drew, Redelings, and
  Cranston}]{mctavish_how_2017}
McTavish, E.~J., B.~T. Drew, B.~Redelings, and K.~A. Cranston. 2017. How and
  {Why} to {Build} a {Unified} {Tree} of {Life}. BioEssays 39.
\bibAnnoteFile{mctavish_how_2017}

\bibitem[{McTavish et~al.(2015)McTavish, Hinchliff, Allman, Brown, Cranston,
  Holder, Rees, and Smith}]{mctavish_phylesystem_2015}
McTavish, E.~J., C.~E. Hinchliff, J.~F. Allman, J.~W. Brown, K.~A. Cranston,
  M.~T. Holder, J.~A. Rees, and S.~A. Smith. 2015. Phylesystem: a git-based
  data store for community-curated phylogenetic estimates. Bioinformatics
  31:2794--2800.
\bibAnnoteFile{mctavish_phylesystem_2015}

\bibitem[{Michonneau et~al.(2016)Michonneau, Brown, and
  Winter}]{michonneau_rotl_2016}
Michonneau, F., J.~W. Brown, and D.~J. Winter. 2016. rotl: an {R} package to
  interact with the {Open} {Tree} of {Life} data. Methods in Ecology and
  Evolution 7:1476--1481.
\bibAnnoteFile{michonneau_rotl_2016}

\bibitem[{Miller et~al.(2020)Miller, McKinnon, Whiting, Kauwe, and
  Ridge}]{miller2020codonpairs}
Miller, J.~B., L.~M. McKinnon, M.~F. Whiting, J.~S. Kauwe, and P.~G. Ridge.
  2020. Codon pairs are phylogenetically conserved: A comprehensive analysis of
  codon pairing conservation across the tree of life. PloS one 15:e0232260.
\bibAnnoteFile{miller2020codonpairs}

\bibitem[{{OpenTreeOfLife} et~al.(2019){OpenTreeOfLife}, {Benjamin Redelings},
  {Luna Luisa Sanchez Reyes}, {Karen A. Cranston}, {Jim Allman}, {Mark T.
  Holder}, and {Emily Jane McTavish}}]{opentreeoflife_open_2019}
{OpenTreeOfLife}, {Benjamin Redelings}, {Luna Luisa Sanchez Reyes}, {Karen A.
  Cranston}, {Jim Allman}, {Mark T. Holder}, and {Emily Jane McTavish}. 2019.
  Open {Tree} of {Life} {Synthetic} {Tree} .
\bibAnnoteFile{opentreeoflife_open_2019}

\bibitem[{{OpenTreeofLife} et~al.(2019){OpenTreeofLife}, {Karen A. Cranston},
  {Benjamin Redelings}, {Luna Luisa Sanchez Reyes}, {Jim Allman}, {Emily Jane
  McTavish}, and {Mark T. Holder}}]{opentreeoflife_open_2019-1}
{OpenTreeofLife}, {Karen A. Cranston}, {Benjamin Redelings}, {Luna Luisa
  Sanchez Reyes}, {Jim Allman}, {Emily Jane McTavish}, and {Mark T. Holder}.
  2019. Open {Tree} of {Life} {Taxonomy} .
\bibAnnoteFile{opentreeoflife_open_2019-1}

\bibitem[{Owen et~al.(2015)Owen, Bracken-Grissom, Stern, and
  Crandall}]{owen2015synthetic}
Owen, C.~L., H.~Bracken-Grissom, D.~Stern, and K.~A. Crandall. 2015. A
  synthetic phylogeny of freshwater crayfish: insights for conservation.
  Philosophical Transactions of the Royal Society B: Biological Sciences
  370:20140009.
\bibAnnoteFile{owen2015synthetic}

\bibitem[{Park et~al.(2018)Park, Worthington, and Xi}]{park2018taxon}
Park, D.~S., S.~Worthington, and Z.~Xi. 2018. Taxon sampling effects on the
  quantification and comparison of community phylogenetic diversity. Molecular
  ecology 27:1296--1308.
\bibAnnoteFile{park2018taxon}

\bibitem[{Parr et~al.(2012)Parr, Guralnick, Cellinese, and
  Page}]{parr_evolutionary_2012}
Parr, C.~S., R.~Guralnick, N.~Cellinese, and R.~D.~M. Page. 2012. Evolutionary
  informatics: unifying knowledge about the diversity of life. Trends in
  Ecology \& Evolution 27:94--103.
\bibAnnoteFile{parr_evolutionary_2012}

\bibitem[{Piel et~al.(2000)Piel, Donoghue, Sanderson, and
  Netherlands}]{piel2000treebase}
Piel, W.~H., M.~Donoghue, M.~Sanderson, and L.~Netherlands. 2000. Treebase: a
  database of phylogenetic information. \emph{in} Proceedings of the 2nd
  International Workshop of Species vol. 2000.
\bibAnnoteFile{piel2000treebase}

\bibitem[{Polilov and Makarova(2017)}]{polilov2017scaling}
Polilov, A.~A. and A.~A. Makarova. 2017. The scaling and allometry of organ
  size associated with miniaturization in insects: A case study for coleoptera
  and hymenoptera. Scientific reports 7:1--7.
\bibAnnoteFile{polilov2017scaling}

\bibitem[{Redelings and Holder(2017)}]{redelings_supertree_2017}
Redelings, B.~D. and M.~T. Holder. 2017. A supertree pipeline for summarizing
  phylogenetic and taxonomic information for millions of species. PeerJ
  5:e3058.
\bibAnnoteFile{redelings_supertree_2017}

\bibitem[{Rees and Cranston(2017)}]{rees_automated_2017}
Rees, J.~A. and K.~Cranston. 2017. Automated assembly of a reference taxonomy
  for phylogenetic data synthesis. Biodiversity Data Journal .
\bibAnnoteFile{rees_automated_2017}

\bibitem[{Rosindell and Harmon(2012)}]{rosindell2012onezoom}
Rosindell, J. and L.~J. Harmon. 2012. Onezoom: a fractal explorer for the tree
  of life. PLoS Biol 10:e1001406.
\bibAnnoteFile{rosindell2012onezoom}

\bibitem[{Sanchez~Reyes et~al.(2020)Sanchez~Reyes, Kandziora, and
  McTavish}]{reyes_physcraper_2020}
Sanchez~Reyes, L.~L., M.~Kandziora, and E.~J. McTavish. 2020. Physcraper: a
  python package for continual update of evolutionary estimates using the
  {Open} {Tree} of {Life}. bioRxiv Page~2020.09.15.299156.
\bibAnnoteFile{reyes_physcraper_2020}

\bibitem[{Sanderson et~al.(2008)Sanderson, Boss, Chen, Cranston, and
  Wehe}]{sanderson2008phylota}
Sanderson, M.~J., D.~Boss, D.~Chen, K.~A. Cranston, and A.~Wehe. 2008. The
  phylota browser: processing genbank for molecular phylogenetics research.
  Systematic Biology 57:335--346.
\bibAnnoteFile{sanderson2008phylota}

\bibitem[{Smith et~al.(2009)Smith, Beaulieu, and Donoghue}]{smith2009mega}
Smith, S.~A., J.~M. Beaulieu, and M.~J. Donoghue. 2009. Mega-phylogeny approach
  for comparative biology: an alternative to supertree and supermatrix
  approaches. BMC evolutionary biology 9:1--12.
\bibAnnoteFile{smith2009mega}

\bibitem[{Smith and Brown(2018)}]{smith2018constructing}
Smith, S.~A. and J.~W. Brown. 2018. Constructing a broadly inclusive seed plant
  phylogeny. American journal of botany 105:302--314.
\bibAnnoteFile{smith2018constructing}

\bibitem[{Smith and Walker(2019)}]{smith2019pyphlawd}
Smith, S.~A. and J.~F. Walker. 2019. Pyphlawd: A python tool for phylogenetic
  dataset construction. Methods in Ecology and Evolution 10:104--108.
\bibAnnoteFile{smith2019pyphlawd}

\bibitem[{Stoltzfus et~al.(2013)Stoltzfus, Lapp, Matasci, Deus, Sidlauskas,
  Zmasek, Vaidya, Pontelli, Cranston, Vos et~al.}]{stoltzfus2013phylotastic}
Stoltzfus, A., H.~Lapp, N.~Matasci, H.~Deus, B.~Sidlauskas, C.~M. Zmasek,
  G.~Vaidya, E.~Pontelli, K.~Cranston, R.~Vos, et~al. 2013. {Phylotastic!
  Making tree-of-life knowledge accessible, reusable and convenient}. BMC
  bioinformatics 14:1--17.
\bibAnnoteFile{stoltzfus2013phylotastic}

\bibitem[{Sukumaran and Holder(2010)}]{sukumaran_dendropy_2010}
Sukumaran, J. and M.~T. Holder. 2010. {DendroPy}: a {Python} library for
  phylogenetic computing. Bioinformatics 26:1569--1571.
\bibAnnoteFile{sukumaran_dendropy_2010}

\bibitem[{Sullivan et~al.(2009)Sullivan, Wood, Iliff, Bonney, Fink, and
  Kelling}]{sullivan_ebird_2009}
Sullivan, B.~L., C.~L. Wood, M.~J. Iliff, R.~E. Bonney, D.~Fink, and
  S.~Kelling. 2009. {eBird}: {A} citizen-based bird observation network in the
  biological sciences. Biological Conservation 142:2282--2292.
\bibAnnoteFile{sullivan_ebird_2009}

\bibitem[{Sánchez-Reyes and O’Meara(2019)}]{sanchez-reyes_datelife_2019}
Sánchez-Reyes, L.~L. and B.~C. O’Meara. 2019. {DateLife}: {Leveraging}
  databases and analytical tools to reveal the dated {Tree} of {Life}. bioRxiv
  Page~782094.
\bibAnnoteFile{sanchez-reyes_datelife_2019}

\bibitem[{Tarka et~al.(2018)Tarka, Guenther, Niemel{\"a}, Nakagawa, and
  Noble}]{tarka2018sex}
Tarka, M., A.~Guenther, P.~T. Niemel{\"a}, S.~Nakagawa, and D.~W. Noble. 2018.
  Sex differences in life history, behavior, and physiology along a slow-fast
  continuum: a meta-analysis. Behavioral ecology and sociobiology 72:1--13.
\bibAnnoteFile{tarka2018sex}

\bibitem[{Uyeda et~al.(2017)Uyeda, Pennell, Miller, Maia, and
  McClain}]{uyeda2017evolution}
Uyeda, J.~C., M.~W. Pennell, E.~T. Miller, R.~Maia, and C.~R. McClain. 2017.
  The evolution of energetic scaling across the vertebrate tree of life. The
  American Naturalist 190:185--199.
\bibAnnoteFile{uyeda2017evolution}

\bibitem[{Wong and Rosindell(2020)}]{wong2020dynamic}
Wong, Y. and J.~Rosindell. 2020. Dynamic visualisation of million-tip trees:
  the onezoom project. bioRxiv .
\bibAnnoteFile{wong2020dynamic}

\end{thebibliography}
% 1. All authors should be listed i.e. no use of et al.
% 2. Dashes should not be used to replace author names in repeat entries
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

% Please include any figure captions on a separate page after the references. Figures themselves should be embedded in the text.

%\begin{figure}[!p]
%\centering\includegraphics{fig1}
%\caption{Figure caption}
%\label{Fig1}
%\end{figure}


%\begin{table}[!p]
% 1. Table titles should be in caps and lowercase
% 2. Footnotes can be used in Tables (a,b,c)}
%\tblcaption{Table title
%\label{Table1}}
%{\tabcolsep=4.25pt
%\begin{tabular}{@{}cccccccccc@{}}
%\tblhead{Heading & Heading & Heading & Heading & Heading}
%Value & Value & Value & Value & Value
%\lastline
%\end{tabular}}
%\end{table}

%If you have any print appendices, please include them at the end of the document.

\end{document}
